src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,965 @@
+            } else {
+                this.x = (gridSize / 2 - 1) * this.scalingFactor
+                this.y = (gridSize / 2) * this.scalingFactor
+            }
+
+            //The amount of numbers to the next turn increases every other turn.
+            this.numberToTurnAtForSpiral = 1
+            this.incrementForNumberToTurnAt = 1
+            this.whetherIncrementShouldIncrement = true
+        }
+    }
+
+    fillGridCell() {
+        this.drawBackgroundSquare()
+
+        this.drawPrimaryColorSquare()
+
+        this.drawSecondaryColorSquare()
+
+        this.showNumber()
+    }
+
+    drawBackgroundSquare() {
+        this.sketch.fill(this.backgroundColor)
+        this.drawBigSquare()
+    }
+
+    drawPrimaryColorSquare() {
+        this.colorMainColorProperties()
+        this.drawBigSquare()
+    }
+
+    drawSecondaryColorSquare() {
+        this.colorSecondaryColorProperties()
+        this.drawSmallSquare()
+    }
+
+    colorMainColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Primary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    colorSecondaryColorProperties() {
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization
+                    === PropertyVisualization.Secondary_Color
+            ) {
+                if (
+                    this.hasProperty(
+                        this.currentNumber,
+                        this.propertyObjects[i].property
+                    )
+                ) {
+                    this.sketch.fill(this.propertyObjects[i].color)
+                }
+            }
+        }
+    }
+
+    drawBigSquare() {
+        this.sketch.rect(
+            this.x,
+            this.y,
+            this.scalingFactor,
+            this.scalingFactor
+        )
+    }
+
+    drawSmallSquare() {
+        this.sketch.rect(
+            this.x + this.scalingFactor / 4,
+            this.y + this.scalingFactor / 4,
+            this.scalingFactor / 2,
+            this.scalingFactor / 2
+        )
+    }
+
+    hasProperty(num: bigint, property: Property) {
+        if (property === Property.Number_Sequence) {
+            for (
+                let a = this.seq.first;
+                a < Math.min(this.seq.last - this.seq.first + 1, 200);
+                a++
```

Comment:
If I choose Random Integers in Range, and then I chose Grid Visualizer and alter the "Property" option for the first property to be "Number_Sequence" then I seem to fall into an infinite loop, even if I decrease the Amount of Numbers to a single digit.  It doesn't look as if this loop here can be the cause because you limit it to 200 iterations, but I'm not sure where else to look for why it hangs if I do that.

Reply:
Number_Sequence puts the natural numbers in a spiral and then highlights the current number sequence.
Any other preset puts the current number sequence in a spiral and then highlights the properties the user selects.
This is a temporary feature because eventually the use should be able to select multiple number sequences with one as the base sequence that goes in a spiral and the rest of the sequences being highlighted if they are contained in the current sequence.

Reply:
See https://github.com/numberscope/frontscope/pull/164#discussion_r1040058421.